includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -589,6 +589,46 @@ public function is_using_saved_payment_method() {
 		return ( isset( $_POST[ 'wc-' . $payment_method . '-payment-token' ] ) && 'new' !== $_POST[ 'wc-' . $payment_method . '-payment-token' ] );
 	}
 
+	/**
+	 * Checks if the payment method is being saved in the current request.
+	 *
+	 * @param stdClass $source
+	 * @param WC_Stripe_Customer $customer
+	 * @param bool $force_save
+	 *
+	 * @return bool
+	 * @since 4.6.0
+	 */
+	public function should_save_payment_method( $source, $customer, $force_save ) {
+		if ( apply_filters( 'wc_stripe_force_save_source', $force_save, $customer ) ) {
+			return true;
+		}
+
+		// The customer has disabled "Saved cards" functionality.
+		if ( ! $this->saved_cards ) {
+			return false;
+		}
+
+		// The payment method in use is already saved.
+		if ( $this->is_using_saved_payment_method() ) {
+			return false;
+		}
+
+		// The customer is not logged in.
```

Comment:
Should this check move ahead of the is_using_saved_payment_method check for a tiny performance improvement?

Reply:
`is_using_saved_payment_method` just checks a few `$_POST` variables, but I'm all for tiny performance improvements :)

Done in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1407/commits/f8e166ae7395ed37012cd01d024d122179447c6b